runtime/lua/hl-yank.lua

```diff
@@ -0,0 +1,26 @@
+local api = vim.api
+
+-- highlight the yanked region with highlight group higroup for timeout ms 
+-- use from init.vim via 
+--   au TextYankPost * lua require'hl_yank'(vim.v.event, 'IncSearch', 500)
+return function(event, higroup, timeout)
+    if event.operator ~= 'y' or event.regtype == '' then return end
+
+    local bn = api.nvim_get_current_buf()
+    local ns = api.nvim_create_namespace('hlyank')
+    api.nvim_buf_clear_namespace(bn, ns, 0, -1)
+
+    local pos1 = vim.fn.getpos("'[")
+    local lin1, col1, off1 = pos1[2] - 1, pos1[3] - 1, pos1[4]
+    local pos2 = vim.fn.getpos("']")
+    local lin2, col2, off2 = pos2[2] - 1, pos2[3] - (event.inclusive and 0 or 1), pos2[4]
+    for l = lin1, lin2 do
+        local c1 = (l == lin1 or event.regtype:byte() == 22) and (col1 + off1) or 0
+        local c2 = (l == lin2 or event.regtype:byte() == 22) and (col2 + off2) or -1
+        api.nvim_buf_add_highlight(bn, ns, higroup, l, c1, c2)
+    end
```

Comment:
my plan was to extract this part to a generic function that takes a pair of marks and returns a "region" (list of row/col pairs). That's something that is sorely missed in Vimscript.

Reply:
I can try -- where should this go?

Reply:
btw there were some [bugs](https://github.com/justinmk/config/blob/47ab07760d6b5eeb38164333931ac35963a64d41/.config/nvim/init.vim#L1104-L1107) in the original logic but we don't need to solve those initially, though it's certainly welcome.

Reply:
if it applies anywhere, it can go directly on `vim`, which would mean `src/nvim/lua/vim.lua` or `runtime/lua/vim/shared.lua`. Maybe `shared` is better

Reply:
> I can try -- where should this go?

probably in `vim.lua` for now. Actually I wanted to add it as an API function but that can be done later I guess.

Reply:
I tested block-visual yank -- it seems to work, except for the case when the visual selection "inverts" (col2 < col1); that should be fixable. Which specific situation did you have in mind in the todo?

Reply:
invert was one case, but main case that bugs me is this:

```
text text text
text text text
text text
```

with `<c-v>$jjy` the highlighted "region" does not include the 3rd "text" column. Since the region is a rectangle (only 2 rol/col pairs), the far column  should perhaps be determined by the longest line of text, rather than where the cursor happens to be placed.

Alternatively the function could return a "jagged" region: _multiple_ rectangles. Then the caller would need to call `nvim_buf_add_highlight` for each rectangle. This would be useful for UIs and various plugins too.

Reply:
Yeah, that's an edge case but not ideal -- if the highlight doesn't actually show what's yanked, one could ask what's the point...

Reply:
As of now, the function doesn't return any rectangles, just lines and column ranges. It just needs (a _lot_) more logic for _which_ columns to return based on the regtype. That will require some thinking...

Reply:
(In fact, the output of `getpos` is already off in these cases -- I don't see at the moment how to deal with that loss of information.)

Reply:
Yeah, the problem is specifically the case of block visual selection where the end mark is on an EOL -- in this case `getpos` returns a useless (can't exactly say "wrong", there may be a good reason for it) result. Not sure how to deal with this :/

Reply:
@archseer I think this function could be useful for your snippet plugin as well?

Reply:
Yeah I have a similar function for coordinates/extmarks: https://github.com/archseer/snippets.nvim/blob/master/lua/snippet/init.lua#L129-L139

I think in the long run there will be extmark decorations (currently used for bufhl?) and extranges, so it'll just be a decoration on the extrange.